### DISCUSSION TITLE
[Tour Editor] Tag images are not displayed

### DISCUSSION THREAD

**Discussion by GurVJ** [mytourbook/mytourbook#1008] [Q&A]
Hello,

with MT 23.3.0 I can't see Tag pictures in the Tour Editor. Am I the only one ?

![Capture 001](https://user-images.githubusercontent.com/64196685/223786524-f2b1bfbf-0547-406a-af60-7ecbc0ac3132.jpg)
![Capture 002](https://user-images.githubusercontent.com/64196685/223786516-74a8dae0-ee9c-4ed5-9216-778d350a6115.jpg)

**Comment by FJBDev** [comment]
I don't see the same issue:

![image](https://user-images.githubusercontent.com/25235665/223883706-bf59ead6-eeae-4539-ad34-58df3a2d193c.png)

Do you see the issue for all your images ?

Can you attach one of the image for which you see this issue so I can try to reproduce it locally ?

**Comment by FJBDev** [comment]
Also, do you see any errors in the log ?

**Comment by GurVJ** [comment]
I see the issue for all the pictures.
![Peregrine 11](https://user-images.githubusercontent.com/64196685/223956648-ade454cd-d9e8-4353-91d3-6395eefd505a.png)
![la-sportiva-akasha-ii-m-chaussures-homme-542704-1-f](https://user-images.githubusercontent.com/64196685/223956652-639b8a9c-a195-4af2-b500-c023b669ff01.jpg)
![main_Saucony_Xodus_Ultra_Trailrunning_schoenen_Heren_2206201045_b149](https://user-images.githubusercontent.com/64196685/223956657-fc0d2d1c-9758-442a-8604-d5d417edb91b.png)

I'll check the logs.

**Reply by GurVJ** [reply]
I found this line dealing with Category Path in the logs ...
![image](https://user-images.githubusercontent.com/64196685/223959145-9975329c-3582-4a7c-b7af-9b2534a29486.png)

**Reply by GurVJ** [reply]
Finally, I am not sure that this warning message is relevant as I also found it few days before installing the 23.3.0 release.

**Reply by FJBDev** [reply]
Thank you for the pictures, I imported them to MT but it seems to be working fine:

![image](https://user-images.githubusercontent.com/25235665/224035041-b05959de-1184-4e3b-bcf5-83bee3eeaf0f.png)

>Finally, I am not sure that this warning message is relevant as I also found it few days before installing the 23.3.0 release.
Correct, this is just a warning

**Comment by GurVJ** [comment]
After some testing, it seems that it is a matter of lenght of Tag names.
There doesn't seem to be a fixed lenght limit but the picture does not display when tag name is more than around 22-26 characters.
@FJBDev , can you test it ?

![image](https://user-images.githubusercontent.com/64196685/224075193-85ef70f5-afa6-43a9-a8ba-6029e11d12c3.png)
![image](https://user-images.githubusercontent.com/64196685/224075310-cd3d9289-431e-4b69-a029-febc803e5581.png)

**Reply by wolfgang-ch** [reply]
I can confirm that when the tag name is too long, then the tag image is not displayed

![image](https://user-images.githubusercontent.com/1283445/224359152-b8b3d574-082c-40e7-b11b-f226d43fafc2.png)

**Reply by wolfgang-ch** [reply]
This is the original stork image before I created a tag image from it

![image](https://user-images.githubusercontent.com/1283445/224359895-a9471e08-2cdb-4127-bf85-610a903aa0e1.png)

**Reply by wolfgang-ch** [reply]
The currently used UI control `org.eclipse.swt.custom.CLabel` which displays the tag image and the text has a limit, when the text is larger than the label max width, then the image is not displayed -> this needs another solution

**Reply by wolfgang-ch** [reply]
> After some testing, it seems that it is a matter of lenght of Tag names. There doesn't seem to be a fixed lenght limit but the picture does not display when tag name is more than around 22-26 characters. @FJBDev , can you test it ?
> 
> ![image](https://user-images.githubusercontent.com/64196685/224075193-85ef70f5-afa6-43a9-a8ba-6029e11d12c3.png) ![image](https://user-images.githubusercontent.com/64196685/224075310-cd3d9289-431e-4b69-a029-febc803e5581.png)

This is fixed in https://github.com/mytourbook/mytourbook/issues/1015#issuecomment-1467547405
